clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/KafkaClientFactory.java

```diff
@@ -0,0 +1,190 @@
+        Objects.requireNonNull(clientSupplier);
+
+        final Promise<T> resultPromise = Promise.promise();
+        createClientWithRetries(bootstrapServersConfig, clientSupplier, getRetriesTimeLimit(retriesTimeout), resultPromise);
+        return resultPromise.future();
+    }
+
+    /**
+     * Creates a new KafkaAdminClient.
+     * <p>
+     * If creation fails because the {@value CommonClientConfigs#BOOTSTRAP_SERVERS_CONFIG} config property contains a
+     * (non-empty) list of URLs that are not (yet) resolvable, further creation attempts are done with a delay of
+     * {@value #CLIENT_CREATION_RETRY_DELAY_MILLIS}ms in between. These retries are done until the given
+     * <em>retriesTimeout</em> has elapsed.
+     *
+     * @param clientConfig The admin client configuration properties.
+     * @param retriesTimeout Thr maximum time for which retries are done. Using a negative duration or {@code null}
+     *                       here is interpreted as an unlimited timeout value.
+     * @return A future indicating the outcome of the creation attempt.
+     * @throws NullPointerException if any of the parameters except retriesTimeout is {@code null}.
+     */
+    public Future<KafkaAdminClient> createKafkaAdminClientWithRetries(
+            final Map<String, String> clientConfig,
+            final Duration retriesTimeout) {
+        Objects.requireNonNull(clientConfig);
+
+        final Promise<KafkaAdminClient> resultPromise = Promise.promise();
+        final String bootstrapServersConfig = clientConfig.get(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG);
+        createClientWithRetries(bootstrapServersConfig, () -> KafkaAdminClient.create(vertx, clientConfig),
+                getRetriesTimeLimit(retriesTimeout), resultPromise);
+        return resultPromise.future();
+    }
+
+    private Instant getRetriesTimeLimit(final Duration retriesTimeout) {
+        if (retriesTimeout == null || retriesTimeout.isNegative()) {
+            return Instant.MAX;
+        }
+        return Instant.now(clock).plus(retriesTimeout);
+    }
+
+    private <T> Future<T> createClientWithRetries(
+            final String bootstrapServersConfig,
+            final Supplier<T> clientSupplier,
+            final Instant retriesTimeLimit,
+            final Promise<T> resultPromise) {
+        try {
+            resultPromise.complete(clientSupplier.get());
+        } catch (final Exception e) {
+            // perform retry in case bootstrap URLs are not resolvable ("No resolvable bootstrap urls given in bootstrap.servers")
+            // (see org.apache.kafka.clients.ClientUtils#parseAndValidateAddresses)
+            if (e instanceof KafkaException
+                    && e.getCause() instanceof ConfigException && e.getCause().getMessage() != null
+                    && e.getCause().getMessage().contains(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG)
+                    && containsValidServerEntries(bootstrapServersConfig)) {
```

Comment:
I've deliberately not matched the whole exception message (or the "No resolvable" part) here for the reason you've given.
However, I have to use some kind of message matching, otherwise I would do retries on *every* kind of `ConfigException`. (No retries should be done if there are issues with other config property values.)
I think matching the name of the property is safe here - this pattern of the exception message containing the config property whose value caused the ConfigException is used for many kinds of such config errors.